scripts/common/check_schema_update.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+
+##--------------------------------------------------------------------
+## Copyright (c) 2018 OSIsoft, LLC
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+##--------------------------------------------------------------------
+
+__author__="Massimiliano Pinto"
+__version__="1.0"
+
+CURRENT_VERSION_FILE=$1
+THIS_VERSION_FILE=$2
+
+# Get installation path as dirname of CURRENT_VERSION_FILE var
+INSTALLED_DIR=$(dirname ${CURRENT_VERSION_FILE})
+# Get new version path as dirname of THIS_VERSION_FILE var
+NEW_VERSION_DIR=$(dirname ${THIS_VERSION_FILE})
+
+#echo "Checking installed foglamp in [${INSTALLED_DIR}] and new installing version [${NEW_VERSION_DIR}]"
+
+# Include common code
+if [ ! "${FOGLAMP_ROOT}" ]; then
+    # Use source path
+    FILE_TO_SOURCE="scripts/common/get_storage_plugin.sh"
```

Comment:
If FOGLAMP_ROOT is not set we default to /usr/local/foglamp - should we not do the same here?

Reply:
- when we do "make install" we have a source tree and FOGLAMP_ROOT should not be set as we refer to this script in the source tree (it's part of the new code, not the one installed)

- when the check is being performed  during the new snap installation the FOGLAMP_ROOT is passed by the post-refresh hook:

export FOGLAMP_ROOT="${SNAP}/usr/local/foglamp" 

and it refers to the new path with new candidate package.